geo/src/test/java/org/apache/carbondata/geo/QuadTreeClsTest.java

```diff
@@ -0,0 +1,317 @@
+        pointList.add(new double[] {18, 11.8});
+        pointList.add(new double[] {11.5, 6.3});
+        pointList.add(new double[] {4.3, 20});
+
+        qtreee.insert(pointList);
+        QuadNode root = qtreee.getRoot();
+
+        Assume.assumeTrue(root.childrenIsNull());
+    }
+
+    /**
+     * Illegal test insertion area
+     */
+    @Test
+    public void testInsertLessThan3Points() throws Exception {
+        List<double[]> pointList = new ArrayList<>();
+        pointList.add(new double[] {4.3, 9.4});
+        pointList.add(new double[] {7.3, 13.8});
+        exception.expect(RuntimeException.class);
+        exception.expectMessage("polygon at least need 4 points, first and last is same.");
+        qtreee.insert(pointList);
+        QuadNode root = qtreee.getRoot();
+        Assume.assumeTrue(root.childrenIsNull());
+    }
+
+    /**
+     * Test creation Hadid
+     */
+    @Test
+    public void testCreateHashID() throws Exception {
+        GridData grid = new GridData(0,0, 15,15,4);
+        // Here is a grid
+        Assume.assumeTrue(grid.createHashID(0,0) == 0);
+        Assume.assumeTrue(grid.createHashID(0,1) == 1);
+        Assume.assumeTrue(grid.createHashID(1,0) == 2);
+        Assume.assumeTrue(grid.createHashID(1,1) == 3);
+
+        Assume.assumeTrue(grid.createHashID(0,2) == 4);
+        Assume.assumeTrue(grid.createHashID(0,3) == 5);
+        Assume.assumeTrue(grid.createHashID(1,2) == 6);
+        Assume.assumeTrue(grid.createHashID(1,3) == 7);
+
+        Assume.assumeTrue(grid.createHashID(2,0) == 8);
+        Assume.assumeTrue(grid.createHashID(2,1) == 9);
+        Assume.assumeTrue(grid.createHashID(3,0) == 10);
+        Assume.assumeTrue(grid.createHashID(3,1) == 11);
+
+        Assume.assumeTrue(grid.createHashID(2,2) == 12);
+        Assume.assumeTrue(grid.createHashID(2,3) == 13);
+        Assume.assumeTrue(grid.createHashID(3,2) == 14);
+        Assume.assumeTrue(grid.createHashID(3,3) == 15);
+        // Center point
+        Assume.assumeTrue(grid.createHashID(0,4) == 16);
+        Assume.assumeTrue(grid.createHashID(0,5) == 17);
+        Assume.assumeTrue(grid.createHashID(1,4) == 18);
+        Assume.assumeTrue(grid.createHashID(1,5) == 19);
+
+        Assume.assumeTrue(grid.createHashID(0,6) == 20);
+        Assume.assumeTrue(grid.createHashID(0,7) == 21);
+        Assume.assumeTrue(grid.createHashID(1,6) == 22);
+        Assume.assumeTrue(grid.createHashID(1,7) == 23);
+
+        Assume.assumeTrue(grid.createHashID(2,4) == 24);
+        Assume.assumeTrue(grid.createHashID(2,5) == 25);
+        Assume.assumeTrue(grid.createHashID(3,4) == 26);
+        Assume.assumeTrue(grid.createHashID(3,5) == 27);
+
+        Assume.assumeTrue(grid.createHashID(2,6) == 28);
+        Assume.assumeTrue(grid.createHashID(2,7) == 29);
+        Assume.assumeTrue(grid.createHashID(3,6) == 30);
+        Assume.assumeTrue(grid.createHashID(3,7) == 31);
+
+
+        System.out.println(grid.createHashID(7,7));
```

Comment:
remove the print statement from the test case, just keep the assert

Reply:
done